File: openapi-test-java-server-quarkus-reactive-client-mp-reactive/testapi/model/Currency.java
package testapi.model;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

/** ISO 4217 currency code */
public record Currency(@JsonValue @JsonCreator String value) {
  public Currency withValue(String value) {
    return new Currency(value);
  };
}

File: openapi-test-scala-server-spring/testapi/model/Currency.scala
package testapi.model

import io.circe.Decoder
import io.circe.Encoder

/** ISO 4217 currency code */
case class Currency(value: String) extends AnyVal

object Currency {
  implicit val decoder: Decoder[Currency] = io.circe.Decoder[java.lang.String].map(Currency.apply)

  implicit val encoder: Encoder[Currency] = io.circe.Encoder[java.lang.String].contramap(_.value)
}